router/core/graphql_prehandler.go

```diff
@@ -655,6 +675,33 @@ func (h *PreHandler) handleOperation(req *http.Request, variablesParser *astjson
 		engineParseSpan.End()
 	}
 
+	if h.accessController != nil {
+		// Based on the authentication result, the introspection config,
+		// and wether this is an introspection query,
+		// we decide here if we need to abort the request or not.
+		isIntrospection, err := operationKit.isIntrospectionQuery()
+		if err != nil {
+			requestContext.logger.Debug("failed to check if operation is introspection, treat it like non-introspection operation", zap.Error(err))
+			isIntrospection = false
+		}
+
+		if httpOperation.authMethod != authenticationMethodNormal {
+			if isIntrospection {
+				if httpOperation.authMethod != authenticationMethodIntrospectionToken && httpOperation.authMethod != authenticationMethodIntrospectionSkip {
+					return &httpGraphqlError{
+						message:    "unauthorized",
+						statusCode: http.StatusUnauthorized,
+					}
+				}
+			} else {
+				return &httpGraphqlError{
+					message:    "unauthorized",
+					statusCode: http.StatusUnauthorized,
+				}
+			}
+		}
+	}
```

Comment:
Can you swap and invert the outer if with the inner one and return early instead of nesting?